### ISSUE TITLE
`extern crate foo as core` breaks resolution in macros

### ISSUE THREAD

**Issue by tmandry** [rust-lang/rust#64364]
Broken in a recent nightly. Reproducer ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=b2ebe9fed3b69a7eaef700018e5a9b51)):

```rust
extern crate futures as core;

fn bar() {
    let _ = format!("{}", 42);
}
```

Gives the following:

```
error[E0433]: failed to resolve: could not find `format_args` in `core`
 --> src/lib.rs:4:13
  |
4 |     let _ = format!("{}", 42);
  |             ^^^^^^^^^^^^^^^^^ could not find `format_args` in `core`
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

I've also observed this if a crate is named `core` and gets used without any `extern crate` items.

modules named `core` do not trigger this, only crates.

Seen in the wild: https://github.com/gfx-rs/gfx/issues/2964

**Comment by petrochenkov** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/63687, fixed in https://github.com/rust-lang/rust/pull/64060.
